service/src/main/java/bio/terra/externalcreds/services/ProviderService.java

```diff
@@ -4,69 +4,72 @@
 import bio.terra.externalcreds.models.LinkedAccountWithPassportAndVisas;
-import bio.terra.externalcreds.models.TokenTypeEnum;
 import com.google.common.annotations.VisibleForTesting;
-import com.nimbusds.jose.JWSHeader;
-import com.nimbusds.jose.shaded.json.JSONObject;
-import com.nimbusds.jwt.JWTParser;
-import java.net.MalformedURLException;
-import java.net.URI;
 import java.sql.Timestamp;
-import java.text.ParseException;
 import java.time.Duration;
 import java.time.Instant;
 import java.time.temporal.ChronoUnit;
 import java.util.Collections;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.Set;
-import java.util.stream.Collectors;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.http.HttpStatus;
 import org.springframework.security.oauth2.client.registration.ClientRegistration;
-import org.springframework.security.oauth2.core.user.OAuth2User;
-import org.springframework.security.oauth2.jwt.Jwt;
-import org.springframework.security.oauth2.jwt.JwtDecoders;
-import org.springframework.security.oauth2.jwt.JwtException;
+import org.springframework.security.oauth2.core.OAuth2AuthorizationException;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
 import org.springframework.web.reactive.function.client.WebClient;
 import reactor.core.publisher.Mono;
 
 @Service
 @Slf4j
 public class ProviderService {
 
-  public static final String PASSPORT_JWT_V11_CLAIM = "passport_jwt_v11";
-  public static final String GA4GH_PASSPORT_V1_CLAIM = "ga4gh_passport_v1";
-  public static final String GA4GH_VISA_V1_CLAIM = "ga4gh_visa_v1";
-  public static final String VISA_TYPE_CLAIM = "type";
-  public static final String JKU_HEADER = "jku";
   public static final String EXTERNAL_USERID_ATTR = "email";
 
   private final ExternalCredsConfig externalCredsConfig;
   private final ProviderClientCache providerClientCache;
   private final OAuth2Service oAuth2Service;
   private final LinkedAccountService linkedAccountService;
+  private final PassportService passportService;
+  private final JwtUtils jwtUtils;
 
   public ProviderService(
       ExternalCredsConfig externalCredsConfig,
       ProviderClientCache providerClientCache,
       OAuth2Service oAuth2Service,
-      LinkedAccountService linkedAccountService) {
+      LinkedAccountService linkedAccountService,
+      PassportService passportService,
+      JwtUtils jwtUtils) {
     this.externalCredsConfig = externalCredsConfig;
     this.providerClientCache = providerClientCache;
     this.oAuth2Service = oAuth2Service;
     this.linkedAccountService = linkedAccountService;
+    this.passportService = passportService;
+    this.jwtUtils = jwtUtils;
   }
 
   public Set<String> getProviderList() {
     return Collections.unmodifiableSet(externalCredsConfig.getProviders().keySet());
   }
 
+  public void refreshExpiringPassports() {
```

Comment:
Any thoughts on whether/how to test this function? 

We tried using `Mockito.verify` to check the number of times that `ProviderService.authAndRefreshPassport` was called, but realized we can't do that because `providerService` would have to be a mock.

Reply:
Thank you! That totally solved the problem. 

We were trying to avoid doing the whole database check because that would involve mocking quite a few things to get `authAndRefreshPassport` to run without error. Also it seemed like better practice to keep this test as more of a "unit" which doesn't re-test any functionality that has already been well tested.